src/core/io/src/4C_io_mesh.hpp

```diff
@@ -39,25 +43,36 @@ namespace Core::IO::MeshInput
   std::string describe(VerbosityLevel level);
 
 
-  struct CellBlock;
-  struct SideSet;
-  struct LineSet;
+  class CellBlock;
   struct PointSet;
 
+  /*!
+   * @brief A point in the mesh
+   */
+  template <unsigned dim>
+  struct Point
+  {
+    /// External ID of the point as defined in the mesh
+    int external_id{};
```

Comment:
1. Should we typedef this? I can see that this might be larger than an `int` someday.
2. Should the index be inside the `Point` or should this be an additional (optional) array in the `Mesh` struct?